pom.xml

```diff
@@ -291,7 +291,7 @@
                             <sourcepath>${project.build.sourceDirectory}:${project.build.directory}/generated-sources/annotations</sourcepath>
                             <show>public</show>
                             <failOnError>true</failOnError>
-                            <failOnWarnings>true</failOnWarnings>
+                            <failOnWarnings>false</failOnWarnings>
```

Comment:
I think we in general prefer to fix the Javadocs instead of ignoring the warnings.

Reply:
The reason I skipped them is: with Java 11, they were already skipped while building. Not sure how. 

There are 58 warnings in total which are all about uncommented methods:

```bash
...
[WARNING] /Users/mabulgu/github-repos/mabulgu/kafka-access-operator/src/main/java/io/strimzi/kafka/access/model/KafkaReference.java:36: warning: no comment
[WARNING] public void setListener(final String listener) {
[WARNING] ^
[WARNING] /Users/mabulgu/github-repos/mabulgu/kafka-access-operator/src/main/java/io/strimzi/kafka/access/model/KafkaReference.java:20: warning: no comment
[WARNING] public void setName(final String name) {
[WARNING] ^
[WARNING] /Users/mabulgu/github-repos/mabulgu/kafka-access-operator/src/main/java/io/strimzi/kafka/access/model/KafkaReference.java:28: warning: no comment
[WARNING] public void setNamespace(final String namespace) {
[WARNING] ^
[WARNING] Generating /Users/mabulgu/github-repos/mabulgu/kafka-access-operator/target/apidocs/io/strimzi/kafka/access/model/KafkaUserReference.html...
[WARNING] /Users/mabulgu/github-repos/mabulgu/kafka-access-operator/src/main/java/io/strimzi/kafka/access/model/KafkaUserReference.java:9: warning: no comment
[WARNING] public class KafkaUserReference {
[WARNING] ^
[WARNING] /Users/mabulgu/github-repos/mabulgu/kafka-access-operator/src/main/java/io/strimzi/kafka/access/model/KafkaUserReference.java:44: warning: no comment
[WARNING] public String getApiGroup() {
[WARNING] ^
[WARNING] /Users/mabulgu/github-repos/mabulgu/kafka-access-operator/src/main/java/io/strimzi/kafka/access/model/KafkaUserReference.java:36: warning: no comment
[WARNING] public String getKind() {
[WARNING] ^
[WARNING] /Users/mabulgu/github-repos/mabulgu/kafka-access-operator/src/main/java/io/strimzi/kafka/access/model/KafkaUserReference.java:19: warning: no comment
[WARNING] public String getName() {
[WARNING] ^
[WARNING] /Users/mabulgu/github-repos/mabulgu/kafka-access-operator/src/main/java/io/strimzi/kafka/access/model/KafkaUserReference.java:27: warning: no comment
[WARNING] public String getNamespace() {
[WARNING] ^
[WARNING] /Users/mabulgu/github-repos/mabulgu/kafka-access-operator/src/main/java/io/strimzi/kafka/access/model/KafkaUserReference.java:48: warning: no comment
[WARNING] public void setApiGroup(final String apiGroup) {
[WARNING] ^
[WARNING] /Users/mabulgu/github-repos/mabulgu/kafka-access-operator/src/main/java/io/strimzi/kafka/access/model/KafkaUserReference.java:40: warning: no comment
[WARNING] public void setKind(final String kind) {
[WARNING] ^
[WARNING] /Users/mabulgu/github-repos/mabulgu/kafka-access-operator/src/main/java/io/strimzi/kafka/access/model/KafkaUserReference.java:23: warning: no comment
[WARNING] public void setName(final String name) {
[WARNING] ^
[WARNING] /Users/mabulgu/github-repos/mabulgu/kafka-access-operator/src/main/java/io/strimzi/kafka/access/model/KafkaUserReference.java:31: warning: no comment
[WARNING] public void setNamespace(final String namespace) {
[WARNING] ^
...
[WARNING] 58 warnings
```

Reply:
I agree. However, without commenting on the methods as expected, the docs are already useless, that was my point and the reason behind skipping the warnings.

I got your point. I will take the main operator as an example and do what's needed here as well, as they should be consistent with these kinda things.

Reply:
Added all required comments on top of the relevant methods and classes.